### PR TITLE
[sdk] Add CustomLayer interface and example

### PR DESCRIPTION
**Pull request by yuryybk**
Resolves https://github.com/mapbox/mapbox-gl-native-android/issues/452

### CODE REVIEW

**Comment by yuryybk** [comment]
> can `example_custom_layer.cpp` be made part of the test application while keeping the .hpp in the sdk? This way it mimics what an end-user would do

indeed, made changes

File: MapboxGLAndroidSDKTestApp/src/main/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+add_library(
+    example-custom-layer MODULE
+    ${PROJECT_SOURCE_DIR}/example_custom_layer.cpp
+)
+
+target_include_directories(
+    example-custom-layer
+    PRIVATE
+        ../../../../MapboxGLAndroidSDK/src/main/cpp
```
Comment (kiryldz) [inline MapboxGLAndroidSDKTestApp/src/main/cpp/CMakeLists.txt:15]: Do we need all cpp files?

Comment (yuryybk) [reply MapboxGLAndroidSDKTestApp/src/main/cpp/CMakeLists.txt:15]: we need to point the whole folder as a public C++ API
for now, it contains only a single custom_layer.hpp, but probably could be extended later

Comment (kiryldz) [reply MapboxGLAndroidSDKTestApp/src/main/cpp/CMakeLists.txt:15]: That's actually my point - if it gets extended we may not want to include it to `MapboxGLAndroidSDKTestApp` automatically. As I understand it 😄

Comment (yuryybk) [reply MapboxGLAndroidSDKTestApp/src/main/cpp/CMakeLists.txt:15]: It is instructions for CMake to build correct paths to search headers, but the usage of every header is the responsibility of user code in the #include section. Seems other headers will not be included in compilation without explicit reference in user code.
As I see it is the best way to manage dependencies, am I wrong?

Comment (kiryldz) [reply MapboxGLAndroidSDKTestApp/src/main/cpp/CMakeLists.txt:15]: Yeah, thought again about it - guess it's valid option here.

**Pull request merged**